### ISSUE TITLE
7$: Disallow labels named 'await'/'yield' in async/generator functions

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#214]

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by singalhimanshu** [comment]
The current error for this code:
```js
function* f() {
  yield:
}
```
Is - 
```shell
main.js:2:8: error: missing semicolon after statement [E027]
main.js:2:8: error: unexpected token [E054]
```
What should be the new errorr?

**Comment by strager** [comment]
> The current error for this code:
> 
> ```js
> function* f() {
>   yield:
> }
> ```
> 
> Is -
> 
> ```shell
> main.js:2:8: error: missing semicolon after statement [E027]
> main.js:2:8: error: unexpected token [E054]
> ```
> 
> What should be the new errorr?

That looks *okay*. It could be better: perhaps "unexpected colon; expected expression or semicolon".

Note: the following code is legal:
```js
function* f() {
  true?yield:null;
}
```
